_posts/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that.md

```diff
@@ -0,0 +1,265 @@
+**Can a large project cope without testing? Probably so. But the number of errors and their severity will probably be much higher.** That’s why in Allegro Ads we pay attention to writing tests.
+
+We have a fairly large number of frameworks that are used to write tests in JavaScript. We can list tools like: [MochaJS](https://mochajs.org/), [Jest](https://jestjs.io/), [Karma](https://karma-runner.github.io), [Jasmine](https://jasmine.github.io/), [Cypress](https://www.cypress.io/). Some of them focus more on business value, others on technical details. There are also a range of libraries which solve different problems. However, in this article, we will pay special attention and compare two testing javascript libraries like [Enzyme.js](https://enzymejs.github.io/enzyme/) and [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) (RTL).
+
+![npm trends screen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/01.png)
+###### Source [npm trends](https://www.npmtrends.com/@testing-library/react-vs-enzyme)
+
+## EnzymeJS
+From the enzyme documentation we can read a short description: *Enzyme is a JavaScript Testing utility for React that makes it easier to test your React Components' output. You can also manipulate, traverse, and in some ways simulate runtime given the output.*
+
+Enzyme was released in **2015 by AirBnB**. And let's face it, Enzyme has gained a lot of popularity. Last year, the library was moved from AirBnB space to external github space as a separate independent organization. This step was intended to encourage the community to further develop the library.
+
+![commit diffrences](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/02.png)
+
+![commit diffrences](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/03.png)
+###### Source [Enzyme github](https://github.com/enzymejs/enzyme/commit/b06750bb5f6248ceb9c3fae903a71d0747a420d6)
+
+When it comes to raw data, the latest version of **Enzyme 3.11.0** has a bundle size at **463.2kb (127.5 kB)** and was released on **20 December 2019**. Currently has 258 [open issues](https://github.com/enzymejs/enzyme/issues) and 31 open PR’s. 	
+
+![bundlephobia enzyme sreen](/img/articles/2021-12-13-why-we-should-rewrite-enzymejs-to-rtl-and-how-to-do-that/04.png)
+###### Source [Bundle Phobia](https://bundlephobia.com/package/enzyme@3.11.0)
+
+
+Interesting fact: 
+- There are also Enzyme Adapter Packages for React. For example, for version 16, it also needs *enzyme-adapter-react-16* to be installed. Unfortunately, it does not support all features completely.
```

Comment:
I think we need some explanation what is the purpose of this adapters. Also it is one of the drawbacks of Enzyme (you have to wait for adapter after new React version) which should be emphazised.